### PR TITLE
Added the _restore function to "unpop" the argument stack.

### PR DESCRIPTION
**Pull request by t-harvey**
This is a pull request based on the discussion #2581.

Thanks!

### CODE REVIEW

**Comment by zherczeg** [comment]
For me this ok, although I haven't reviewed the original code. Perhaps @akosthekiss could help.

File: jerry-ext/arg/arg-js-iterator-helper.c [hunk 1]
```diff
@@ -30,6 +30,28 @@ jerryx_arg_js_iterator_pop (jerryx_arg_js_iterator_t *js_arg_iter_p) /**< the JS
...
+ * @return the restored (now current) JS argument.
+ */
+jerry_value_t
+jerryx_arg_js_iterator_restore (jerryx_arg_js_iterator_t *js_arg_iter_p) /**< the JS arg iterator */
+{
+  if (js_arg_iter_p->js_arg_idx == 0)
+  {
+    return jerry_create_undefined ();
+  }
+  else
```
Comment (LaszloLango) [inline jerry-ext/arg/arg-js-iterator-helper.c]: Minor style issue: the `else` is not needed, because of the `return` statement.

**Comment by LaszloLango** [comment]
Please add some test into the `tests/unit-ext/test-ext-arg.c` file.

**Comment by zherczeg** [comment]
I think another round of review could be done here.

**Comment by LaszloLango** [comment]
Sorry for the late response. I am not sure this is right way to do this, but I am not against this PR. Please add the new API function to the related documentation (`docs/09.EXT-REFERENCE-ARG.md`). The patch is good to me after that.

File: tests/unit-ext/test-ext-arg.c [hunk 2]
```diff
@@ -561,6 +564,152 @@ test_validator_array2_handler (const jerry_value_t func_obj_val __attribute__((u
...
+ * jerryx_arg_js_iterator_restore function.  We'll introduce a union
+ * type to hold a bool or double and a transform function that will
+ * look for this type.  Then, we'll call the handler with two
+ * parameters, one bool and one double and see if we correctly build
+ * the union types for each parameter.  To check that the code protects
+ * against backing up too far, when the check for the double fails,
+ * we'll "restore" the stack three times; this shouldn't break
+ * anything.
+*/
+typedef enum
```
Comment (zherczeg) [inline tests/unit-ext/test-ext-arg.c]: Please add a short description to these two structs and their members.

File: tests/unit-ext/test-ext-arg.c [hunk 3]
```diff
@@ -561,6 +564,152 @@ test_validator_array2_handler (const jerry_value_t func_obj_val __attribute__((u
...
+ * of the installed functions, like jerryx_arg_bool().
+ */
+#define jerryx_arg_double_or_bool(value_ptr, coerce_or_not, optional_or_not, last_parameter) \
+        jerryx_arg_custom (value_ptr, \
+                           (uintptr_t)&((uintptr_t []){(uintptr_t)coerce_or_not, \
+                                                       (uintptr_t)optional_or_not, \
+                                                       (uintptr_t)last_parameter}), \
+                           jerry_arg_to_double_or_bool)
+
+static jerry_value_t
```
Comment (zherczeg) [inline tests/unit-ext/test-ext-arg.c]: This function also needs a description.

**Pull request merged**